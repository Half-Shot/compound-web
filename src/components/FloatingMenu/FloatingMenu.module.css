/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.menu {
  border-radius: var(--cpd-space-3x);
  background: var(--cpd-color-bg-canvas-default);

  /* Here we use outline to create a border internal to the container */
  outline: var(--cpd-border-width-1) solid
    var(--cpd-color-border-interactive-secondary);
  outline-offset: calc(-1 * var(--cpd-border-width-1));
  box-shadow: 0 4px 24px 0 rgba(0 0 0 / 10%);
  inline-size: fit-content;
  display: flex;
  flex-direction: column;
  gap: var(--cpd-space-1x);
  padding-block: var(--cpd-space-5x) var(--cpd-space-4x);
}

.title {
  color: var(--cpd-color-text-secondary);
  padding-inline: var(--cpd-space-4x);
  padding-block-end: calc(var(--cpd-space-2x) - var(--cpd-border-width-1));
  border-block-end: var(--cpd-border-width-1) solid var(--cpd-color-gray-400);
  margin-block: 0 var(--cpd-space-2x);
}
