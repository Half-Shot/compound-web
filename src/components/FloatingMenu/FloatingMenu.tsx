/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import classnames from "classnames";
import React, { ComponentPropsWithoutRef, ReactNode, forwardRef } from "react";
import styles from "./FloatingMenu.module.css";
import useId from "../../utils/useId";
import { Text } from "../Typography/Text";

interface Props extends ComponentPropsWithoutRef<"div"> {
  /**
   * The size of the menu in px.
   */
  // TODO: Figure out whether this component should come with a standard set of
  // T-shirt sizes
  size: number;
  /**
   * The menu title.
   */
  title: string;
  /**
   * The CSS class.
   */
  className?: string;
  /**
   * The menu contents.
   */
  children: ReactNode;
}

/**
 * A floating menu (the kind of menu you would see on desktop).
 */
// This an internal component not intended for export! Consumers should use it
// via the Menu or ContextMenu components (which at the time of writing, are not
// yet built).
export const FloatingMenu = forwardRef<HTMLDivElement, Props>(
  ({ size, title, className, children, ...props }, ref) => {
    const titleId = useId();
    return (
      <div
        role="menu"
        ref={ref}
        aria-labelledby={titleId}
        className={classnames(className, styles.menu)}
        style={{ inlineSize: size }}
        {...props}
      >
        <Text
          as="h3"
          id={titleId}
          className={styles.title}
          size="sm"
          weight="semibold"
        >
          {title}
        </Text>
        {children}
      </div>
    );
  },
);

FloatingMenu.displayName = "FloatingMenu";
