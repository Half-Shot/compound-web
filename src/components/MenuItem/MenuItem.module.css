/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.item {
  display: flex;
  align-items: center;
  padding-block: var(--cpd-space-2x);
  padding-inline: var(--cpd-space-4x);
  box-sizing: border-box;
  inline-size: 100%;
  color: var(--cpd-color-text-secondary);
  background: var(--cpd-color-bg-action-secondary-rest);
}

.item.interactive {
  cursor: pointer;
}

.icon {
  flex-shrink: 0;
}

.label {
  padding-inline: var(--cpd-space-3x);
  flex-basis: 100%;
  text-align: start;
}

.nav-hint {
  display: none;
  flex-shrink: 0;
  margin-inline-end: calc(-1 * var(--cpd-space-2x));
}

button.item {
  appearance: none;
  border: none;
}

.item.with-chevron {
  padding-inline-end: var(--cpd-space-2x);
}

.item[data-kind="primary"] > .label {
  color: var(--cpd-color-text-primary);
}

.item[data-kind="primary"] > .icon {
  color: var(--cpd-color-icon-primary);
}

.item[data-kind="primary"] > .nav-hint {
  color: var(--cpd-color-icon-tertiary);
}

.item[data-kind="critical"] > .label {
  color: var(--cpd-color-text-critical-primary);
}

.item[data-kind="critical"] > .icon,
.item[data-kind="critical"] > .nav-hint {
  color: var(--cpd-color-icon-critical-primary);
}

@media (hover) {
  .item.interactive[data-kind="primary"]:hover {
    background: var(--cpd-color-bg-action-secondary-hovered);
  }

  .item.interactive[data-kind="critical"]:hover {
    background: var(--cpd-color-bg-critical-subtle);
  }

  .item.interactive:hover > .nav-hint {
    display: initial;
  }

  .item.interactive:hover > .nav-hint ~ * {
    display: none;
  }
}

.item.interactive[data-kind="primary"]:active {
  background: var(--cpd-color-bg-action-secondary-pressed);
}

.item.interactive[data-kind="critical"]:active {
  background: var(--cpd-color-bg-critical-subtle);
}
