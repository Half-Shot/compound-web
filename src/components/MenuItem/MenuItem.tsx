/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import classnames from "classnames";
import React, {
  ComponentPropsWithoutRef,
  ComponentType,
  ElementType,
  SVGAttributes,
} from "react";
import styles from "./MenuItem.module.css";
import { Text } from "../Typography/Text";
import ChevronRightIcon from "@vector-im/compound-design-tokens/icons/chevron-right.svg";

type MenuItemElement = "button" | "label" | "div";

type Props<C extends MenuItemElement> = {
  /**
   * The element type of this menu item.
   * @default button
   */
  as?: C;
  className?: string;
  /**
   * The icon to show on this menu item.
   */
  Icon: ComponentType<SVGAttributes<SVGElement>>;
  /**
   * The label to show on this menu item.
   */
  label: string;
  /**
   * The color variant of the menu item.
   * @default primary
   */
  kind?: "primary" | "critical";
  /**
   * Whether to replace the children with a navigation hint on hover.
   * @default true
   */
  navHint?: boolean;
} & ComponentPropsWithoutRef<C>;

export const MenuItem = <C extends MenuItemElement = "button">({
  as,
  className,
  Icon,
  label,
  kind = "primary",
  navHint = true,
  children,
  ...props
}: Props<C>) => {
  const Component = as ?? ("button" as ElementType);

  return (
    <Component
      {...props}
      className={classnames(className, styles.item, {
        [styles.interactive]: as !== "div",
      })}
      data-kind={kind}
    >
      <Icon width={24} height={24} className={styles.icon} aria-hidden={true} />
      <Text className={styles.label} size="md" weight="medium" as="span">
        {label}
      </Text>
      {navHint && (
        <ChevronRightIcon
          width={24}
          height={24}
          className={styles["nav-hint"]}
          aria-hidden={true}
        />
      )}
      {children}
    </Component>
  );
};
