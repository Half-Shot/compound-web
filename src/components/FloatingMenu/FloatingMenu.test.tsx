/*
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import React from "react";
import UserProfileIcon from "@vector-im/compound-design-tokens/icons/user-profile.svg";
import NotificationsIcon from "@vector-im/compound-design-tokens/icons/notifications.svg";
import ChatProblemIcon from "@vector-im/compound-design-tokens/icons/chat-problem.svg";
import LeaveIcon from "@vector-im/compound-design-tokens/icons/leave.svg";

import { FloatingMenu } from "./FloatingMenu";
import { MenuItem } from "../MenuItem/MenuItem";
import { MenuDivider } from "../MenuItem/MenuDivider";

describe("Menu", () => {
  it("renders", () => {
    const { asFragment } = render(
      <FloatingMenu size={200} title="Settings">
        <MenuItem Icon={UserProfileIcon} label="Profile" />
        <MenuItem Icon={NotificationsIcon} label="Notifications" />
        <MenuItem Icon={ChatProblemIcon} label="Feedback" />
        <MenuDivider />
        <MenuItem kind="critical" Icon={LeaveIcon} label="Sign out" />
      </FloatingMenu>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
