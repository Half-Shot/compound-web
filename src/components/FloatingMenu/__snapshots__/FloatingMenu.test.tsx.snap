// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Menu > renders 1`] = `
<DocumentFragment>
  <div
    aria-labelledby=":r0:"
    class="_menu_2a2b6b"
    role="menu"
    style="inline-size: 200px;"
  >
    <h3
      class="_font-body-sm-semibold_489030 _title_2a2b6b"
      id=":r0:"
    >
      Settings
    </h3>
    <button
      class="_item_37eef5 _interactive_37eef5"
      data-kind="primary"
      role="menuitem"
    >
      <svg
        aria-hidden="true"
        class="_icon_37eef5"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.175 13.825C9.95833 14.6083 10.9 15 12 15C13.1 15 14.0417 14.6083 14.825 13.825C15.6083 13.0417 16 12.1 16 11C16 9.9 15.6083 8.95833 14.825 8.175C14.0417 7.39167 13.1 7 12 7C10.9 7 9.95833 7.39167 9.175 8.175C8.39167 8.95833 8 9.9 8 11C8 12.1 8.39167 13.0417 9.175 13.825ZM13.4125 12.4125C13.0208 12.8042 12.55 13 12 13C11.45 13 10.9792 12.8042 10.5875 12.4125C10.1958 12.0208 10 11.55 10 11C10 10.45 10.1958 9.97917 10.5875 9.5875C10.9792 9.19583 11.45 9 12 9C12.55 9 13.0208 9.19583 13.4125 9.5875C13.8042 9.97917 14 10.45 14 11C14 11.55 13.8042 12.0208 13.4125 12.4125Z"
          fill="currentColor"
        />
        <path
          d="M22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12ZM20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12Z"
          fill="currentColor"
        />
        <path
          d="M16.2297 18.7917C15.7472 18.6085 15.2623 18.4571 14.775 18.3375C13.8583 18.1125 12.9333 18 12 18C11.0667 18 10.1417 18.1125 9.225 18.3375C8.73774 18.4571 8.25283 18.6085 7.77028 18.7917C7.12696 18.3902 6.54483 17.8998 6.04104 17.3378C6.93176 16.9259 7.83474 16.6092 8.75 16.3875C9.81667 16.1292 10.9 16 12 16C13.1 16 14.1833 16.1292 15.25 16.3875C16.1653 16.6092 17.0682 16.9259 17.959 17.3378C17.4552 17.8998 16.873 18.3902 16.2297 18.7917Z"
          fill="currentColor"
        />
      </svg>
      <span
        class="_font-body-md-medium_489030 _label_37eef5"
      >
        Profile
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_37eef5"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.69999 17.3C8.51665 17.1167 8.42499 16.8834 8.42499 16.6C8.42499 16.3167 8.51665 16.0834 8.69999 15.9L12.6 12L8.69999 8.10005C8.51665 7.91672 8.42499 7.68338 8.42499 7.40005C8.42499 7.11672 8.51665 6.88338 8.69999 6.70005C8.88332 6.51672 9.11665 6.42505 9.39999 6.42505C9.68332 6.42505 9.91665 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8708 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8708 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91665 17.4834 9.68332 17.575 9.39999 17.575C9.11665 17.575 8.88332 17.4834 8.69999 17.3Z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="_item_37eef5 _interactive_37eef5"
      data-kind="primary"
      role="menuitem"
    >
      <svg
        aria-hidden="true"
        class="_icon_37eef5"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3C19 3 19 10 19 10V16L20.2929 17.2929C20.9229 17.9229 20.4767 19 19.5858 19H4.41424C3.52333 19 3.07716 17.9229 3.70713 17.2929L5.00002 16V10C5.00002 10 5.00002 3 12 3ZM17 10.0104L16.9994 9.98802C16.9985 9.96033 16.9963 9.91049 16.9914 9.84171C16.9816 9.70348 16.9608 9.49294 16.9179 9.23505C16.8303 8.70978 16.659 8.04 16.3362 7.39443C16.0173 6.75669 15.5704 6.17979 14.9375 5.75785C14.3171 5.34426 13.4001 5 12 5C10.6 5 9.68293 5.34426 9.06255 5.75785C8.42964 6.17979 7.98275 6.75669 7.66388 7.39443C7.34109 8.04 7.16973 8.70978 7.08218 9.23505C7.0392 9.49294 7.01849 9.70348 7.00861 9.84171C7.0037 9.91049 7.00154 9.96033 7.00062 9.98802L7.00002 10.0104V16.8284L6.82845 17H17.1716L17 16.8284V10.0104Z"
          fill="currentColor"
        />
        <path
          d="M12 22C10.8954 22 10 21.1046 10 20H14C14 21.1046 13.1046 22 12 22Z"
          fill="currentColor"
        />
      </svg>
      <span
        class="_font-body-md-medium_489030 _label_37eef5"
      >
        Notifications
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_37eef5"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.69999 17.3C8.51665 17.1167 8.42499 16.8834 8.42499 16.6C8.42499 16.3167 8.51665 16.0834 8.69999 15.9L12.6 12L8.69999 8.10005C8.51665 7.91672 8.42499 7.68338 8.42499 7.40005C8.42499 7.11672 8.51665 6.88338 8.69999 6.70005C8.88332 6.51672 9.11665 6.42505 9.39999 6.42505C9.68332 6.42505 9.91665 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8708 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8708 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91665 17.4834 9.68332 17.575 9.39999 17.575C9.11665 17.575 8.88332 17.4834 8.69999 17.3Z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="_item_37eef5 _interactive_37eef5"
      data-kind="primary"
      role="menuitem"
    >
      <svg
        aria-hidden="true"
        class="_icon_37eef5"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.7125 16.7125C12.5208 16.9042 12.2833 17 12 17C11.7167 17 11.4792 16.9042 11.2875 16.7125C11.0958 16.5208 11 16.2833 11 16C11 15.7167 11.0958 15.4792 11.2875 15.2875C11.4792 15.0958 11.7167 15 12 15C12.2833 15 12.5208 15.0958 12.7125 15.2875C12.9042 15.4792 13 15.7167 13 16C13 16.2833 12.9042 16.5208 12.7125 16.7125Z"
          fill="currentColor"
        />
        <path
          d="M12.7125 12.7125C12.5208 12.9042 12.2833 13 12 13C11.7167 13 11.4792 12.9042 11.2875 12.7125C11.0958 12.5208 11 12.2833 11 12V8C11 7.71667 11.0958 7.47917 11.2875 7.2875C11.4792 7.09583 11.7167 7 12 7C12.2833 7 12.5208 7.09583 12.7125 7.2875C12.9042 7.47917 13 7.71667 13 8V12C13 12.2833 12.9042 12.5208 12.7125 12.7125Z"
          fill="currentColor"
        />
        <path
          d="M2.95002 16.3C2.63335 15.6167 2.39585 14.9167 2.23752 14.2C2.07918 13.4833 2.00002 12.75 2.00002 12C2.00002 10.6167 2.26252 9.31667 2.78752 8.1C3.31252 6.88333 4.02502 5.825 4.92502 4.925C5.82502 4.025 6.88335 3.3125 8.10002 2.7875C9.31668 2.2625 10.6167 2 12 2C13.3833 2 14.6834 2.2625 15.9 2.7875C17.1167 3.3125 18.175 4.025 19.075 4.925C19.975 5.825 20.6875 6.88333 21.2125 8.1C21.7375 9.31667 22 10.6167 22 12C22 13.3833 21.7375 14.6833 21.2125 15.9C20.6875 17.1167 19.975 18.175 19.075 19.075C18.175 19.975 17.1167 20.6875 15.9 21.2125C14.6834 21.7375 13.3833 22 12 22C11.25 22 10.5167 21.9208 9.80002 21.7625C9.08335 21.6042 8.38335 21.3667 7.70002 21.05L2.75002 22.5C2.36668 22.6167 2.03335 22.5333 1.75002 22.25C1.46668 21.9667 1.38335 21.6333 1.50002 21.25L2.95002 16.3ZM7.15002 19.1C7.23335 19.0667 7.32502 19.0458 7.42502 19.0375C7.52502 19.0292 7.61668 19.025 7.70002 19.025C7.85002 19.025 7.99585 19.0375 8.13752 19.0625C8.27918 19.0875 8.41668 19.1333 8.55002 19.2C9.08335 19.4667 9.64168 19.6667 10.225 19.8C10.8083 19.9333 11.4 20 12 20C14.2333 20 16.125 19.225 17.675 17.675C19.225 16.125 20 14.2333 20 12C20 9.76667 19.225 7.875 17.675 6.325C16.125 4.775 14.2333 4 12 4C9.76668 4 7.87502 4.775 6.32502 6.325C4.77502 7.875 4.00002 9.76667 4.00002 12C4.00002 12.6 4.06668 13.1917 4.20002 13.775C4.33335 14.3583 4.53335 14.9167 4.80002 15.45C4.91668 15.6667 4.97918 15.8958 4.98752 16.1375C4.99585 16.3792 4.96668 16.6167 4.90002 16.85L3.95002 20.05L7.15002 19.1Z"
          fill="currentColor"
        />
      </svg>
      <span
        class="_font-body-md-medium_489030 _label_37eef5"
      >
        Feedback
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_37eef5"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.69999 17.3C8.51665 17.1167 8.42499 16.8834 8.42499 16.6C8.42499 16.3167 8.51665 16.0834 8.69999 15.9L12.6 12L8.69999 8.10005C8.51665 7.91672 8.42499 7.68338 8.42499 7.40005C8.42499 7.11672 8.51665 6.88338 8.69999 6.70005C8.88332 6.51672 9.11665 6.42505 9.39999 6.42505C9.68332 6.42505 9.91665 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8708 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8708 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91665 17.4834 9.68332 17.575 9.39999 17.575C9.11665 17.575 8.88332 17.4834 8.69999 17.3Z"
          fill="currentColor"
        />
      </svg>
    </button>
    <hr
      class="_divider_2f054a"
    />
    <button
      class="_item_37eef5 _interactive_37eef5"
      data-kind="critical"
      role="menuitem"
    >
      <svg
        aria-hidden="true"
        class="_icon_37eef5"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 12.9999C14.2833 12.9999 14.5208 12.904 14.7125 12.7124C14.9042 12.5207 15 12.2832 15 11.9999C15 11.7165 14.9042 11.479 14.7125 11.2874C14.5208 11.0957 14.2833 10.9999 14 10.9999C13.7167 10.9999 13.4792 11.0957 13.2875 11.2874C13.0958 11.479 13 11.7165 13 11.9999C13 12.2832 13.0958 12.5207 13.2875 12.7124C13.4792 12.904 13.7167 12.9999 14 12.9999Z"
          fill="currentColor"
        />
        <path
          d="M10.385 21.7879C10.142 21.5989 10 21.3079 10 20.9999V2.99987C10.0006 2.84816 10.0356 2.69858 10.1024 2.56237C10.1692 2.42615 10.2661 2.30687 10.3856 2.21349C10.5052 2.12011 10.6444 2.05507 10.7927 2.02327C10.941 1.99146 11.0947 1.99372 11.242 2.02987L19.242 4.02987C19.4584 4.08385 19.6506 4.20865 19.788 4.38441C19.9253 4.56017 19.9999 4.77681 20 4.99987V18.9999C20.0001 19.223 19.9255 19.4397 19.7882 19.6155C19.6508 19.7913 19.4585 19.916 19.242 19.9699L11.242 21.9699C11.0947 22.0068 10.941 22.0096 10.7924 21.9781C10.6439 21.9465 10.5046 21.8815 10.385 21.7879ZM18 5.78088L12 4.28088V19.7189L18 18.2189V5.78088Z"
          fill="currentColor"
        />
        <path
          d="M9 5.99988H7L7 17.9999H9V19.9999H7C5.89543 19.9999 5 19.1044 5 17.9999V5.99988C5 4.89531 5.89543 3.99988 7 3.99988H9V5.99988Z"
          fill="currentColor"
        />
      </svg>
      <span
        class="_font-body-md-medium_489030 _label_37eef5"
      >
        Sign out
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_37eef5"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.69999 17.3C8.51665 17.1167 8.42499 16.8834 8.42499 16.6C8.42499 16.3167 8.51665 16.0834 8.69999 15.9L12.6 12L8.69999 8.10005C8.51665 7.91672 8.42499 7.68338 8.42499 7.40005C8.42499 7.11672 8.51665 6.88338 8.69999 6.70005C8.88332 6.51672 9.11665 6.42505 9.39999 6.42505C9.68332 6.42505 9.91665 6.51672 10.1 6.70005L14.7 11.3C14.8 11.4 14.8708 11.5084 14.9125 11.625C14.9542 11.7417 14.975 11.8667 14.975 12C14.975 12.1334 14.9542 12.2584 14.9125 12.375C14.8708 12.4917 14.8 12.6 14.7 12.7L10.1 17.3C9.91665 17.4834 9.68332 17.575 9.39999 17.575C9.11665 17.575 8.88332 17.4834 8.69999 17.3Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
